apps/www/components/ui/code-editor.tsx

```diff
@@ -0,0 +1,37 @@
+import { Highlight, type PrismTheme } from "prism-react-renderer";
+
+export function CodeEditor({
+    codeBlock,
+    language,
+    theme,
+  }: { codeBlock: string; language: string; theme?: PrismTheme }) {
+    return (
+      <Highlight theme={theme} code={codeBlock} language={language}>
+        {({ tokens, getLineProps, getTokenProps }) => {
+          const amountLines = tokens.length;
+          const gutterPadLength = Math.max(String(amountLines).length, 2);
+          return (
+            <pre
+              key={codeBlock} // Use codeBlock as a key to trigger animations on change
```

Comment:
The added complexity which may be caused by hashing overhead may not yield significant performance benefits. Given the current size of our code blocks, it might be best to stick with the straightforward approach and avoid unnecessary overhead. @chronark What do you think?

Reply:
agreed @unrenamed